tools/wavetool.py

```diff
@@ -0,0 +1,279 @@
+#! /usr/bin/python3
+
+import os
+import sys
+import argparse
+import numpy as np
+import scipy.signal as signal
+import scipy.io.wavfile as wavefile
+import scipy.fftpack as fftpack
+
+eps = 0.0000000001 # used to avoid zero division error
+
+class WavGenerator:
+    @staticmethod
+    def generateSine(param_dict):
+        time = np.arange(0, float(param_dict['duration']), 1.0/float(param_dict['sample_rate']), dtype=np.float32)
+        data = float(param_dict['amp']) * np.sin(2 * np.pi * float(param_dict['freq']) * time + float(param_dict['phase']))
+        return np.reshape(np.repeat(data, int(param_dict['channel'])),[len(data), int(param_dict['channel'])])
+
+    @staticmethod
+    def generateCosine(param_dict):
+        time = np.arange(0, float(param_dict['duration']), 1.0/float(param_dict['sample_rate']), dtype=np.float32)
+        data = float(param_dict['amp']) * np.cos(2 * np.pi * float(param_dict['freq']) * time + float(param_dict['phase']))
+        return np.reshape(np.repeat(data, int(param_dict['channel'])),[len(data), int(param_dict['channel'])])
+
+    @staticmethod
+    def generateRandom(param_dict):
+        wave_size = int(float(param_dict['duration']) * float(param_dict['sample_rate']))
+        data = np.random.normal(float(param_dict['freq']), float(param_dict['phase']), wave_size)
+        maxnum = abs(np.max(data))
+        if abs(np.max(data)) < abs(np.min(data)):
+            maxnum = abs(np.min(data))
+        data = float(param_dict['amp']) / maxnum * data
+        return np.reshape(np.repeat(data, int(param_dict['channel'])),[len(data), int(param_dict['channel'])])
+
+if __name__ == "__main__":
+    def parse_cmdline():
```

Comment:
There's no reason not to have these functions at the top-level. In fact they must be at the top-level for interactive debug or importing https://docs.python.org/3/library/__main__.html